blog/config/common/di/rbac.php

```diff
@@ -9,7 +9,13 @@
 use Yiisoft\Rbac\Php\AssignmentsStorage;
 use Yiisoft\Rbac\Php\ItemsStorage;
 
-/** @var array $params */
+/** 
+ * @var array $params
+ * @var array $params['yiisoft/aliases']
+ * @var array $params['yiisoft/aliases']['aliases']
+ * @var string $params['yiisoft/aliases']['aliases']['@root'] 
```

Comment:
Does that work in PhpStorm? Elsewhere?

Reply:
Just on wampserver at the moment now with windows 11.  Have not tested on phpStorm yet.

Reply:
I mean that is IDE annotation and, as far as I know, no IDE supports this style.

Reply:
The following code gives no psalm level 1 errors and is reasonably explicit. 

````
<?php

declare(strict_types=1);

use Yiisoft\Access\AccessCheckerInterface;
use Yiisoft\Rbac\AssignmentsStorageInterface;
use Yiisoft\Rbac\ItemsStorageInterface;
use Yiisoft\Rbac\Manager;
use Yiisoft\Rbac\Php\AssignmentsStorage;
use Yiisoft\Rbac\Php\ItemsStorage;

/**
 * @see $params['yiisoft/aliases']['aliases']['@root']
 * @var array $params
 * @var string $root
 */

$root = array_search('@root', $params);
return [
    ItemsStorageInterface::class => [
        'class' => ItemsStorage::class,
        '__construct()' => [
            'directory' => $root . DIRECTORY_SEPARATOR . 'resources' . DIRECTORY_SEPARATOR . 'rbac',
        ],
    ],
    AssignmentsStorageInterface::class => [
        'class' => AssignmentsStorage::class,
        '__construct()' => [
            'directory' => $root . DIRECTORY_SEPARATOR . 'resources' . DIRECTORY_SEPARATOR . 'rbac',
        ],
    ],
    AccessCheckerInterface::class => Manager::class,
];

````